tests/unit/admin/test-class-wc-rest-payments-accounts-controller.php

```diff
@@ -0,0 +1,129 @@
+		wp_set_current_user( 1 );
+
+		$this->mock_api_client = $this->createMock( WC_Payments_API_Client::class );
+		$this->controller      = new WC_REST_Payments_Accounts_Controller( $this->mock_api_client );
+
+		// Inject the mocked client into service (happens in WCPay int, so reflection here).
+		$account_service     = WC_Payments::get_account_service();
+		$property_reflection = ( new ReflectionClass( $account_service ) )->getProperty( 'payments_api_client' );
+		$property_reflection->setAccessible( true );
+		$this->original_api_client = $property_reflection->getValue( $account_service );
+		$property_reflection->setValue( $account_service, $this->mock_api_client );
+	}
+
+	public function tearDown() {
+		parent::tearDown();
+
+		// Restore the original client.
+		$account_service     = WC_Payments::get_account_service();
+		$property_reflection = ( new ReflectionClass( $account_service ) )->getProperty( 'payments_api_client' );
+		$property_reflection->setAccessible( true );
+		$property_reflection->setValue( $account_service, $this->original_api_client );
+	}
+
+	public function test_get_account_data_with_connected_account() {
+		$this->mock_api_client
+			->expects( $this->once() )
+			->method( 'is_server_connected' )
+			->willReturn( true );
+		$this->mock_api_client
+			->expects( $this->once() )
+			->method( 'get_account_data' )
+			->willReturn(
+				// We are providing only some of fields, needed for the assertions we are relying to.
+				[
+					'country'          => 'DE',
+					'status'           => 'complete',
+					'store_currencies' => [ 'default' => 'EUR' ],
+				]
+			);
+
+		$response      = $this->controller->get_account_data( new WP_REST_Request( 'GET' ) );
+		$response_data = $response->get_data();
+
+		$this->assertSame( 200, $response->status );
+		$this->assertSame( 'complete', $response_data['status'] );
+		$this->assertSame( 'DE', $response_data['country'] );
+		$this->assertSame( 'EUR', $response_data['store_currencies']['default'] );
+	}
+
+	public function test_get_account_data_without_connected_account_and_enabled_onboarding() {
+		$this->mock_api_client
+			->expects( $this->once() )
+			->method( 'is_server_connected' )
+			->willReturn( true );
+		$this->mock_api_client
+			->expects( $this->once() )
+			->method( 'get_account_data' )
+			->willReturn(
+				// Indicates that server connection is ok, but no connected accounts available.
+				[]
+			);
+
+		$response      = $this->controller->get_account_data( new WP_REST_Request( 'GET' ) );
+		$response_data = $response->get_data();
+
+		$this->assertSame( 200, $response->status );
+		$this->assertSame( 'NOACCOUNT', $response_data['status'] );
+		// The default country and currency have changed in WC 5.3, hence multiple options in assertions.
+		$this->assertContains( $response_data['country'], [ 'US', 'GB' ] );
+		$this->assertContains( $response_data['store_currencies']['default'], [ 'USD', 'GBP' ] );
```

Comment:
can we assert for other `default` parameters as well:
In case a PR touches them in the future, we'd want to make sure the default retains the same value.
'card_present_eligible' , 'current_deadline' , 'has_overdue_requirements', 'has_pending_requirements' etc...

Reply:
Other defaults are not playing any role, as mobile apps will rely on account status in the first place. Hence so few assertions.

Reply:
Oh I see, I thought the expected response from 991-gh-Automattic/woocommerce-payments-server#issue-950384265 will be used somewhere in the future by mobile apps.